azkaban-common/src/test/java/azkaban/executor/container/watch/KubernetesWatchTest.java

```diff
@@ -267,6 +273,17 @@ public void testFlowStatusManagerListenerTransitionCompleted() throws Exception
     // Verify the Pod deletion API is invoked.
     verify(updatingListener.getContainerizedImpl()).deleteContainer(EXECUTION_ID_WITH_SUCCEESS);
 
+    // Verify pod statuses are handled by ContainerStatusMetricsHandlerListener to emit status
+    //metrics. In total there are 10 events, of which Scheduled and InitContainersRunning are
+    //duplicated event statuses.
```

Comment:
I recommend creating a new test method for this. The current method is already quite packed. In general, a complex test method makes it harder to understand the intention of the test  debug any failures.

Reply:
good idea.